ext/inferred_proxy_headers.c

```diff
@@ -0,0 +1,45 @@
+#include "inferred_proxy_headers.h"
+
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
+
+static HashTable proxy_info_map;
+
+void ddtrace_add_proxy_info(const char *system, const char *span_name, const char *component) {
+    ddtrace_proxy_info *info = emalloc(sizeof(ddtrace_proxy_info));
+    info->span_name = span_name;
+    info->component = component;
+    zend_hash_str_add_ptr(&proxy_info_map, system, strlen(system), info);
+}
+
+void ddtrace_init_proxy_info_map(void) {
+    zend_hash_init(&proxy_info_map, 8, NULL, NULL, 1);
+
+    ddtrace_add_proxy_info("aws-apigateway", "aws.apigateway", "aws-apigateway");
+
+    // Add more proxies using ddtrace_add_proxy_info
+}
+
+void ddtrace_free_proxy_info_map(void) {
+    ddtrace_proxy_info *info;
+    ZEND_HASH_FOREACH_PTR(&proxy_info_map, info) {
+        efree(info);
+    } ZEND_HASH_FOREACH_END();
```

Comment:
Nit: the idiomatic way would be using:
```
static dd_proxy_info_dtor(zval *zv) {
    efree(Z_PTR_P(zv));
}
```
and passing it to zend_hash_init as destructor.